### PR TITLE
#110: parameter support for getImageData

### LINKED ISSUE #110: CanvasRenderingContext2D#getImageData should check and use parameters
Original `CanvasRenderingContext2D#getImageData` requires 4 or 5 parameters, and uses width/height parameters for the size of the returned `ImageData` object.

### PR DESCRIPTION
**Pull request by danielrentz**
fixes #110

- check parameter count for `CanvasRenderingContext2d#getImageData`
- use width/height parameter for resulting `ImageData`
- adapt unit test

### CODE REVIEW

File: src/classes/CanvasRenderingContext2D.js [hunk 1]
```diff
@@ -1053,8 +1053,16 @@ export default class CanvasRenderingContext2D {
     return this._fontStack[this._stackIndex];
   }
 
-  getImageData() {
-    return new ImageData(this._canvas.width, this._canvas.height);
+  getImageData(x, y, w, h) {
+    if (arguments.length < 4)
+      throw new TypeError(
```
Comment (hustcc) [inline src/classes/CanvasRenderingContext2D.js:1058]: it will throw error in real api？

Comment (danielrentz) [reply src/classes/CanvasRenderingContext2D.js:1058]: Yes. You can try it in the browser console:
```
document.createElement("canvas").getContext("2d").getImageData()
```
(tested in Firefox, Chrome, Edge)

However, I will add a test for width/height parameters, method will throw if either is zero.

**Review by hustcc — APPROVED** [review]
